packages/@aws-cdk/aws-servicediscovery/lib/service.ts

```diff
@@ -0,0 +1,224 @@
+import cdk = require('@aws-cdk/cdk');
+import { INamespace, NamespaceType } from './namespace';
+import { CfnService} from './servicediscovery.generated';
+
+export interface ServiceProps {
+
+  /**
+   * The ID of the namespace that you want to use for DNS configuration.
+   */
+  namespace: INamespace;
+
+  /**
+   * The DNS type of the record that you want AWS Cloud Map to create. Supported record types
+   * include A, AAAA, A_AAAA, CNAME, and SRV.
+   */
+  dnsType: DnsRecordType;
+
+  /**
+   * The amount of time, in seconds, that you want DNS resolvers to cache the settings for this
+   * record.
+   *
+   * @default 60
+   */
+  dnsTtlSec?: number;
+
+  /**
+   * Public DNS namespaces only. A complex type that contains settings for an optional health check.
+   * If you specify settings for a health check, AWS Cloud Map associates the health check with the
+   * records that you specify in DnsConfig.
+   */
+  healthCheckConfig?: HealthCheckConfig;
+
+  /**
+   * Structure containing failure threshold for a custom health checker.
+   * See: https://docs.aws.amazon.com/cloud-map/latest/api/API_HealthCheckCustomConfig.html
+   */
+  healthCheckCustomConfig?: HealthCheckCustomConfig;
+
+  /**
+   * The routing policy that you want to apply to all DNS records that AWS Cloud Map creates when you
+   * register an instance and specify this service.
+   *
+   * @default WEIGHTED for CNAME records, MULTIVALUE otherwise
+   */
+  routingPolicy?: RoutingPolicy;
+
+  /**
+   * A name for the Service.
+   *
+   * @default CloudFormation-generated name
+   */
+  name?: string;
+
+  /**
+   * A description of the service.
+   *
+   * @default none
+   */
+  description?: string;
+}
+
+/**
+ * Define a CloudMap Service
+ */
+export class Service extends cdk.Construct {
+  /**
+   * A name for the Cloudmap Service.
+   */
+  public readonly name: string;
+
+  public readonly namespaceId: string;
+
+  /**
+   *  The namespace for the Cloudmap Service.
+   */
+  public readonly namespace: INamespace;
+
+  /**
+   * The ID of the namespace that you want to use for DNS configuration.
+   */
+  public readonly serviceArn: string;
+
+  constructor(scope: cdk.Construct, id: string, props: ServiceProps) {
+    super(scope, id);
+
+    const routingPolicy = props.dnsType && props.dnsType === DnsRecordType.Cname
+      ? RoutingPolicy.Weighted
+      : RoutingPolicy.Multivalue;
+
+    const dnsType = props.dnsType !== undefined ? props.dnsType : DnsRecordType.A;
+
+    const service = new CfnService(this, 'Resource', {
+      name: props.name,
+      description: props.description,
+      dnsConfig: props.namespace.type === NamespaceType.Http ? undefined :
+      {
+        dnsRecords: [
+          {
+            type: dnsType,
```

Comment:
Validate that dnsType is set when you make it optioanl.

Reply:
Wait, namespaceId (and routingPolicy) also both seem to not be set in case of an HTTP namespace.

How can this work?